### ISSUE TITLE
Unable to plot an image on a canvas in Safari Version 17.2

### ISSUE THREAD

**Issue by baogorek** [r-wasm/webr#479]
Safari Version 17.2 (19617.1.17.11.9)

This example, https://github.com/baogorek/Miscellaneous/blob/master/R-code/hello-webR.html, works in Chrome and Edge. What you should see is 
![image](https://github.com/user-attachments/assets/288afa45-5b8c-4d95-9077-68e7e3af69dc)

But if you see nothing after "Hello, World! x = 12," then you've reproduced the problem. 

I know some of these messages are not related to the error ([Issue 189](https://github.com/r-wasm/webr/issues/189)). 
![image](https://github.com/user-attachments/assets/167c0cc3-4472-45a1-853f-f4a0effcbb4d)

It appears that the problem is in the way that the image is being handled and sent to the canvas.

```
      const rCode2 = `
          webr::install("ggplot2")
          library(ggplot2)
          webr::canvas()
          x <- rnorm(100)
          y <- rnorm(100)
          df <- data.frame(x=x, y=y)
          # Plot with axes, points, and labels all in one frame
          p <- ggplot(df, aes(x=x, y=y)) + geom_point()
          print(p)
          dev.off()
      `;

      await webR.evalRVoid(rCode2);

        for (;;) {
          const output = await webR.read();
          switch (output.type) {
            case 'canvas':
              if (output.data.event === 'canvasImage') {
                canvas.getContext('2d').drawImage(output.data.image, 0, 0);
              } else if (output.data.event === 'canvasNewPage') {
                 console.log("Not doing anything");
              }
              break;
            default:
              console.log(output);
          }
        }
```

Let me know if I can be of further help.

**Comment by georgestagg** [comment]
Please could you try again with the latest webR build? Currently I am unable to reproduce, the image generates OK for me right now.

![Screenshot 2024-09-27 at 14 55 28](https://github.com/user-attachments/assets/1d3500f7-d10d-4e24-b0a7-f187d0e182ff)

**Comment by baogorek** [comment]
@georgestagg since the script is importing from 'https://webr.r-wasm.org/latest/webr.mjs', using the latest version should be automatic, right?

I am still getting the same error this morning, but this is a corporate controlled Mac and I'm wondering if there are some extra security measures in place

Any other diagnostic ideas?

**Comment by georgestagg** [comment]
> since the script is importing from 'https://webr.r-wasm.org/latest/webr.mjs', using the latest version should be automatic, right?

Yes, I pushed an update and wondered if the change I made was related. I suppose it is not.

> I am still getting the same error this morning, but this is a corporate controlled Mac and I'm wondering if there are some extra security measures in place

Interesting! Perhaps... The only error I have not seen before is the `NotReadableError`. The cross-origin errors are not ideal, but if it were a breaking issue you would not see the result of `x = 12` in text output.

I wonder if `OffScreenCanvas` is not available on your Safari. Could you please open a console in devtools and show me the output of running `OffScreenCanvas`. If it is undefined or some other error occurs, then Canvas graphics won't work on your machine - but I can show you a workaround if that is indeed the case.

**Comment by baogorek** [comment]
Hi @georgestagg , I do see that OffscreenCanvas is defined:

![image](https://github.com/user-attachments/assets/2cbb6c1b-d3ba-4ab9-8bab-10b46bfa603d)
